### ISSUE TITLE
Shake event not working on micro:bit v1.3?

### ISSUE THREAD

**Issue by microbit-giles** [microbit-foundation/microbit-firmata#11]
In testing the development code.org AppLab app, I can't get 'on shake' events to work on a v1.3 micro:bit (with separate compass and accelerometer chips) but they do work on a 1.5 board (combined single compass and accelerometer).
Is there something in Firmata that could explain this and be fixed?

**Comment by carlosperate** [comment]
There shoudn't be a reason that doesn't work, is using DAL and when this was released micro:bit v1.3 and v1.5 were both fully supported.
Do the events trigger with a normal MakeCode application in that micro:bit?

**Comment by microbit-giles** [comment]
Yes a normal MakeCode project with 'on shake' works fine on the same board.

```
# DAPLink Firmware - see https://mbed.com/daplink
Unique ID: 9900000032044e4500148008000000650000000097969901
HIC ID: 97969901
Auto Reset: 1
Automation allowed: 0
Overflow detection: 0
Daplink Mode: Interface
Interface Version: 0250
Git SHA: 682d8303e37355532402b8d93c4f240a3cec02a9
Local Mods: 0
USB Interfaces: MSD, CDC, HID, WebUSB
Interface CRC: 0x3f2b7e12
Remount count: 1
URL: https://microbit.org/device/?id=9900&v=0250
```

**Comment by microbit-carlos** [comment]
@microbit-giles it's been a while but do you know if this issue is still present?

**Comment by microbit-giles** [comment]
@microbit-carlos No idea I'm afraid - not had chance to look at this in a long time.

**Comment by martinwork** [comment]
@microbit-carlos I did a quick test with microbit-firmata-v1.1.hex, and mbTests.js. My impression was that 1.3, 1.5 and 2.0 all behaved about the same. 1.3 with microbit-firmata-v1.0.hex felt similar.

It seemed that not every shake caused the shake counter to increment. There were perhaps some 1.3/1.5 differences in the number of times I saw 3G, and I noticed "undefined" events with 2.0.

By contrast, with a MakeCode program that counts shakes in "on shake", and displays the count in forever, it seemed easier to generate a shake, and felt more like it counted every shake.

**Comment by microbit-carlos** [comment]
Thanks for the testing Martin, that's very interesting.

I assume the "on shake" MakeCode block is subscribing to the shake event that CODAL triggers, which I think this project does as well, not sure why the results would be different.

Either way, it sounds like the difference is not too significant, so we can probably park this and return to it if we find the difference to be more pronounced.
